pinot-common/src/test/java/org/apache/pinot/common/metrics/BrokerJMXToPromMetricsTest.java

```diff
@@ -0,0 +1,162 @@
+    jmxReporter.start();
+
+    _httpClient = new HttpClient();
+  }
+
+  @Test
+  public void timerTest() {
+    //first assert on global timers
+    Stream.of(BrokerTimer.values()).filter(BrokerTimer::isGlobal)
+        .peek(timer -> _brokerMetrics.addTimedValue(timer, 30_000, TimeUnit.MILLISECONDS))
+        .forEach(timer -> assertTimerExportedCorrectly(timer.getTimerName(), EXPORTED_METRIC_PREFIX));
+    //Assert on local timers
+    Stream.of(BrokerTimer.values()).filter(timer -> !timer.isGlobal()).peek(timer -> {
+      _brokerMetrics.addTimedTableValue(LABEL_VAL_RAW_TABLENAME, timer, 30_000L, TimeUnit.MILLISECONDS);
+    }).forEach(timer -> assertTimerExportedCorrectly(timer.getTimerName(), EXPORTED_LABELS_FOR_RAW_TABLE_NAME,
+        EXPORTED_METRIC_PREFIX));
+  }
+
+  @Test
+  public void gaugeTest() {
+    //global gauges
+    Stream.of(BrokerGauge.values()).filter(BrokerGauge::isGlobal)
+        .peek(gauge -> _brokerMetrics.setOrUpdateGlobalGauge(gauge, () -> 5L))
+        .forEach(gauge -> assertGaugeExportedCorrectly(gauge.getGaugeName(), EXPORTED_METRIC_PREFIX));
+    //local gauges
+    Stream.of(BrokerGauge.values()).filter(gauge -> !gauge.isGlobal()).peek(gauge -> {
+      if (gauge == BrokerGauge.REQUEST_SIZE) {
+        _brokerMetrics.setOrUpdateTableGauge(LABEL_VAL_RAW_TABLENAME, gauge, 5L);
+      } else {
+        _brokerMetrics.setOrUpdateTableGauge(TABLE_NAME_WITH_TYPE, gauge, 5L);
+      }
+    }).forEach(gauge -> {
+      if (gauge == BrokerGauge.REQUEST_SIZE) {
+        assertGaugeExportedCorrectly(gauge.getGaugeName(), EXPORTED_LABELS_FOR_RAW_TABLE_NAME, EXPORTED_METRIC_PREFIX);
+      } else {
+        assertGaugeExportedCorrectly(gauge.getGaugeName(), EXPORTED_LABELS_FOR_TABLE_NAME_TABLE_TYPE,
+            EXPORTED_METRIC_PREFIX);
+      }
+    });
+  }
+
+  @Test
+  public void meterTest() {
+
+    List<BrokerMeter> globalMetersWithExceptionsPrefix =
+        List.of(BrokerMeter.UNCAUGHT_GET_EXCEPTIONS, BrokerMeter.UNCAUGHT_POST_EXCEPTIONS,
+            BrokerMeter.QUERY_REJECTED_EXCEPTIONS, BrokerMeter.REQUEST_COMPILATION_EXCEPTIONS,
+            BrokerMeter.RESOURCE_MISSING_EXCEPTIONS);
+
+    Stream.of(BrokerMeter.values()).filter(BrokerMeter::isGlobal)
+        .peek(meter -> _brokerMetrics.addMeteredGlobalValue(meter, 5L)).forEach(meter -> {
+          if (globalMetersWithExceptionsPrefix.contains(meter)) {
+            String exportedMeterPrefix = String.format("%s_%s", EXPORTED_METRIC_PREFIX_EXCEPTIONS,
+                StringUtils.remove(meter.getMeterName(), "Exceptions"));
+            assertMeterExportedCorrectly(exportedMeterPrefix, EXPORTED_METRIC_PREFIX);
+          } else {
+            assertMeterExportedCorrectly(meter.getMeterName(), EXPORTED_METRIC_PREFIX);
+          }
+        });
+
+    List<BrokerMeter> localMetersThatAcceptRawTableName =
+        List.of(BrokerMeter.QUERIES, BrokerMeter.NO_SERVER_FOUND_EXCEPTIONS, BrokerMeter.DOCUMENTS_SCANNED,
```

Comment:
If we can encode this information in the metric definition, then we can remove the hardcoded list.

Reply:
TBH I think this can be done in this PR. Just edit the enum to add a new method that returns the name. The method should be implemented by default returning the default name/label and only in these cases override to return the particular, not standard name/label

Reply:
@gortiz My view here is that the enums should be decoupled with the _exported_ names and labels. The exported names are property of the exporter's config  and not the enums themselves.

Further, `localMetersThatAcceptRawTableName` basically captures the current usage of the metric. Someone can very well supply `tableNameWithType` to this meter tomorrow and it would still be correct usage. I don't think enums should define what attributes can be attached to them.

@soumitra-st

Reply:
Let's plan to standardize the metric definition in subsequent PRs, then we can remove the hardcoded list. We can traverse the definition and figure out different class of metrics. This will ensure no metrics are missed in testing, and no additional work would be required while adding new metrics.

Reply:
Problem is, to maintain backward compatibility with these non-standard exported names, we might have to support them forever, OR we might break dashboard/alerts of anyone using these metrics. But standardization is definitely required going ahead.

Reply:
I have some reservations about this about how metrics are being used (or abused) in code, will discuss this.